tools/publisher/src/package.rs

```diff
@@ -86,6 +87,26 @@ impl Package {
     pub fn locally_depends_on(&self, other: &PackageHandle) -> bool {
         self.local_dependencies.contains(other)
     }
+
+    /// Returns the expected owners of the crate.
+    pub fn expected_owners(&self) -> HashSet<String> {
+        let mut ret = HashSet::new();
+
+        // Crate ownership for SDK crates. Crates.io requires that at least one owner
+        // is an individual rather than a team, so we use the automation user for that.
+        ret.insert(String::from(RUST_SDK_CI_OWNER));
+
+        if self.category.is_sdk() {
+            ret.insert(String::from(RUST_SDK_OWNER));
+        } else if self.handle.name == "aws-smithy-http-server" {
```

Comment:
This check really should live in the package category code since it's already missing the `aws-smithy-http-server-python` crate. We probably shouldn't add a new category since that will require changes in a lot of places, but we could add a `is_server` function (and potentially store some data on the package category to make that determination).

Reply:
Didn't realize we were already publishing `aws-smithy-http-server-python` automatically too. I've addressed this in https://github.com/awslabs/smithy-rs/pull/1619/commits/6cf564d60db39ec36832df3c67ff48012deeace4.

> This check really should live in the package category code

The problem is the determination of whether a crate should be owned by the server group is not based on package category, but on the crate name. If we put the check there we'd need to store data like the crate name, or data derived from it, on `PackageCategory`. I'm not sure that's the right modeling. It feels more intuitive to model `PackageCategory` and crate owner as orthogonal concepts, and map from the former to the latter in this `expected_owners` function.